src/cmd/support_archive/logs.go

```diff
@@ -0,0 +1,125 @@
+package support_archive
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const logCollectorName = "logCollector"
+
+type logCollector struct {
+	collectorCommon
+
+	context context.Context
+	pods    v1.PodInterface
+}
+
+func newLogCollector(context context.Context, log logr.Logger, supportArchive tarball, pods v1.PodInterface) collector { //nolint:revive // argument-limit doesn't apply to constructors
+	return logCollector{
+		collectorCommon: collectorCommon{
+			log:            log,
+			supportArchive: supportArchive,
+		},
+		context: context,
+		pods:    pods,
+	}
+}
+
+func (collector logCollector) Do() error {
+	logInfof(collector.log, "Starting log collection")
+
+	podList, err := collector.getPodList()
+	if err != nil {
+		return err
+	}
+
+	podGetOptions := metav1.GetOptions{}
+
+	for _, podItem := range podList.Items {
+		pod, err := collector.pods.Get(collector.context, podItem.Name, podGetOptions)
+		if err != nil {
+			logErrorf(collector.log, err, "Unable to get pod info for %s", podItem.Name)
+		} else {
+			collector.collectPodLogs(pod)
+		}
+	}
+	return nil
+}
+
+func (collector logCollector) Name() string {
+	return logCollectorName
+}
+
+func (collector logCollector) getPodList() (*corev1.PodList, error) {
+	listOptions := metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "pod",
+		},
+		LabelSelector: "app.kubernetes.io/name=dynatrace-operator",
```

Comment:
I actually like to not re-use consts in test in these occasions, because it automatically makes sure, that the const isn't involuntarily changed. It is important that the label selector is kept correct. If it changes that test should fail and then force a thought process.

Reply:
we have these constants in the code already, please use them from `kubeobjects.labels.go`